backend/src/main/java/com/bakdata/conquery/models/events/CBlock.java

```diff
@@ -388,4 +408,78 @@ else if (min == Integer.MAX_VALUE) {
 
 		return in.span(span);
 	}
+
+
+	public <T> Int2ObjectMap<T> getColumnIndex(Column column){
+		return (Int2ObjectMap<T>) columnIndices.get(column);
+	}
+
+	public static Int2ObjectMap<?> createColumnIndex(Column column, Bucket bucket) {
+
+		final Int2ObjectMap<Object> index = new Int2ObjectAVLTreeMap<>();
+
+		for (int entity : bucket.entities()) {
+			final int entityEnd = bucket.getEntityEnd(entity);
+			Object entityIndex = index.get(entity);
+
+			for (int event = bucket.getEntityStart(entity); event < entityEnd; event++) {
+				if (bucket.has(event, column)){
+					entityIndex = indexColumn(bucket, event, column, entityIndex);
+				}
+			}
+
+			if (entityIndex != null) {
+				index.put(entity, entityIndex);
+			}
+		}
```

Comment:
Das ist pro Column-Bucket ein Index. Kann es dann vorkommen, dass du beim `for (int entity : bucket.entities())` zwei mal `Object entityIndex = index.get(entity);` für die selbe Entity aufrufst?

Reply:
Pro Import in dem die Entity vorkommt wird es aufgerufen werden. (Hier ist der Unterschied wichtig zwischen Bucket (der Zahl) und Bucket (dem Slice eins Imports) wichtig)